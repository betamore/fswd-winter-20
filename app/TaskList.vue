<template>
    <ol>
        <Task v-for="task in tasks" :key="task.id" :task="task" />
    </ol>
</template>

<script>
import Task from "./Task.vue";
import axios from "axios";

export default {
    components: { Task },
    data: function() {
        return { tasks: [] };
    },
    async mounted() {
        const response = await axios.get("http://localhost:8000/tasks");
        this.tasks = response.data;
    }
};
</script>
