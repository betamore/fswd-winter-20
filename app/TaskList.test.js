/**
 * @jest-environment jsdom
 */

import { mount, createLocalVue } from "@vue/test-utils";
import Vuex from "vuex";
import TaskList from "./TaskList.vue";

const localVue = createLocalVue();

localVue.use(Vuex);

describe("Task vue component", function() {
    it("should render something", function() {
        const wrapper = mount(TaskList, {
            stubs: ["router-link", "router-view"],
            store: new Vuex.Store({
                state: {
                    tasks: [{ id: 1234, name: "Task name #1" }]
                }
            }),
            localVue
        });
        expect(wrapper.element).toMatchSnapshot();
    });
});
