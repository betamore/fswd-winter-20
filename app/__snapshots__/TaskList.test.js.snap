// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Task vue component should render something 1`] = `
<ol>
  <li>
    <router-link-stub
      to="/tasks/1234"
    >
       Task name #1
    </router-link-stub>
     
    <span>
      is completed
    </span>
  </li>
</ol>
`;
